src/processor.ts

```diff
@@ -25,10 +28,22 @@ const transformRecord = ({ recordId, data }: EventRecord): ResultRecord => {
       metadata: { partitionKeys: { serviceDay: serviceDayKey(datetime) } },
     };
   } catch (error) {
+    maybeCaptureError(error);
     return { recordId, result: "ProcessingFailed", data };
   }
 };
 
+// Only capture an error once per run; if many records throw errors and we try
+// to capture them all, the capturing itself can take a long time and cause the
+// Lambda to time out
```

Comment:
question: will an exception getting thrown cause the lambda to restart? Seems like we don't want to lose a lot of errors, because it could be a bug in the lambda and not just a problem with the record.

Reply:
Since we're catching all exceptions within the processing of each record and instead returning `ProcessingFailed` for that record, the Lambda will not stop or restart when this happens. The point of this once-per-run code is to intentionally "lose" all such errors other than the first, because otherwise the Lambda times out and and we don't see anything at all in Sentry (unknown why we don't at least get a few errors; possibly the separate error-sending process never gets a chance to run, as JS is single-threaded).

Assuming we could report more than one error per run, I'm not sure what the value of this would be. Could you elaborate on what information you're concerned we might lose?

Reply:
> Well, if processing different records fails for different reasons we might want to know what they are.

We should be able to determine when this happens by looking in the `failed` prefix in the S3 bucket, as every record we assign a status of `ProcessingFailed` will be dumped in there. So Sentry can alert us to the fact that there's at least one issue, and then we can look at S3 to see all the issues (or, all the records that caused issues, but we should be able to reproduce the issues from there).

> Is this just a limitation of the Firehose model?

We could increase the timeout for the Lambda — the default is 3 seconds, but Firehose supports a timeout of up to 5 minutes. If you think it's worth trying, with some additional Terraform work, we could do so.

Reply:
> So Sentry can alert us to the fact that there's at least one issue, and then we can look at S3 to see all the issues (or, all the records that caused issues, but we should be able to reproduce the issues from there).

Yeah, if we're able to run them through this to reproduce, that should be enough.